### PR TITLE
Update to inliner cost model information.

### PR DESCRIPTION
**Pull request by stuartarchibald**
This adds the call expr that is requesting inlining to
the information provided to the cost model. This is so it
is possible to e.g. decide to inline based on the location of
the call with respect to the rest of the block/IR.

### CODE REVIEW

**Comment by seibert** [comment]
This has a conflict with master now.

**Comment by stuartarchibald** [comment]
@seibert thanks, resolved.

File: numba/tests/test_ir_inlining.py [hunk 1]
```diff
@@ -328,10 +328,14 @@ def impl():
 
     def test_inlining_models(self):
 
-        def s17_caller_model(caller_info, callee_info):
+        def s17_caller_model(expr, caller_info, callee_info):
+            self.assertTrue(isinstance(expr, ir.Expr))
```
Comment (sklam) [inline numba/tests/test_ir_inlining.py]:
```suggestion
            self.assertIsInstance(expr, ir.Expr)
```

File: numba/tests/test_ir_inlining.py [hunk 2]
```diff
@@ -328,10 +328,14 @@ def impl():
...
+            self.assertEqual(expr.op, "call")
             return self.sentinel_17_cost_model(caller_info)
 
-        def s17_callee_model(caller_info, callee_info):
+        def s17_callee_model(expr, caller_info, callee_info):
+            self.assertTrue(isinstance(expr, ir.Expr))
```
Comment (sklam) [inline numba/tests/test_ir_inlining.py]:
```suggestion
            self.assertIsInstance(expr, ir.Expr)
```

File: numba/tests/test_ir_inlining.py [hunk 3]
```diff
@@ -642,10 +646,14 @@ def impl():
 
     def test_inlining_models(self):
 
-        def s17_caller_model(caller_info, callee_info):
+        def s17_caller_model(expr, caller_info, callee_info):
+            self.assertTrue(isinstance(expr, ir.Expr))
```
Comment (sklam) [inline numba/tests/test_ir_inlining.py]:
```suggestion
            self.assertIsInstance(expr, ir.Expr)
```

File: numba/tests/test_ir_inlining.py [hunk 4]
```diff
@@ -642,10 +646,14 @@ def impl():
...
+            self.assertEqual(expr.op, "call")
             return self.sentinel_17_cost_model(caller_info.func_ir)
 
-        def s17_callee_model(caller_info, callee_info):
+        def s17_callee_model(expr, caller_info, callee_info):
+            self.assertTrue(isinstance(expr, ir.Expr))
```
Comment (sklam) [inline numba/tests/test_ir_inlining.py]:
```suggestion
            self.assertIsInstance(expr, ir.Expr)
```

**Comment by stuartarchibald** [comment]
@sklam thanks for the review, fixes in c9738aa.

**Pull request merged**